insights/specs/datasources/yum_updates.py

```diff
@@ -18,83 +16,176 @@ def sorted_cmp(it, cmp):
     sorted_cmp = sorted
 
 
-class UpdatesManager:
-    """ Performs package resolution on yum based systems """
-    def __init__(self):
+# Select which manager to use based on the available system libraries.
+try:
+    try:
```

Comment:
Would it be possible to move the class definitions out of the try catches, instead set a variable? Also I noticed there is some duplicate code in the UpdatesManager class definitions. Would it be possible to have a base class that the classes inherit, that way the duplicate functions and variables can be defined in the base class ex: __init__, __enter__, __exit__, pkg_nevra? Or could you just use one class, and perform checks in each function base on the variable set in the try catches?

Reply:
Simplified imports and changed it into 2 classes, one inheriting from the other.

Reply:
Thank you that looks better.